src/Effects/EffectMgr.cpp

```diff
@@ -130,29 +132,29 @@ static const rtosc::Ports local_ports = {
                 if (val>=0) {
                     eff->numerator = val;
                     int Pdelay, Pfreq;
-                    float freq;
-                    if(eff->denominator) {
+                    float freq, delay;
+                    if (eff->numerator&&eff->denominator) {
+                        eff->efx->speedfactor = (float)eff->denominator / (4.0f *(float)eff->numerator);
                         switch(eff->nefx) {
                         case 2: // Echo
+                        case 10: // Reverse
                             // invert:
-                            // delay = (Pdelay / 127.0f * 1.5f); //0 .. 1.5 sec
-                            Pdelay = (int)roundf((20320.0f / (float)eff->time->tempo) * 
-                                                 ((float)eff->numerator / (float)eff->denominator));
-                            if (eff->numerator&&eff->denominator)
-                                eff->seteffectparrt(2, Pdelay);
+                            // delay = ((Pdelay+1)/128.0f*MAX_REV_DELAY_SECONDS); //0 .. x sec
+                            // Pdelay = (delay * 128.0f / MAX_REV_DELAY_SECONDS) -1
+                            // delay = 60 / tempo * 4 * numerator / denominator
+                            delay = 60.0f / ((float)eff->time->tempo * eff->efx->speedfactor);
```

Comment:
Should we add an assert that this will never divide by zero?

Reply:
ok